bb8/src/inner.rs

```diff
@@ -83,13 +83,36 @@ where
     }
 
     pub(crate) async fn get(&self) -> Result<PooledConnection<'_, M>, RunError<M::Error>> {
+        self.get_with(|this, conn| PooledConnection::new(this, conn))
+            .await
+    }
+
+    pub(crate) async fn get_owned(
+        &self,
+    ) -> Result<PooledConnection<'static, M>, RunError<M::Error>> {
+        self.get_with(|this, conn| {
+            let pool = PoolInner {
+                inner: Arc::clone(&this.inner),
+            };
+            PooledConnection::new_owned(pool, conn)
+        })
+        .await
+    }
+
+    pub(crate) async fn get_with<'a, 'b, F>(
+        &'a self,
+        make_pool: F,
```

Comment:
Let's call this `make_pooled`.

Reply:
You used `make_pool` rather than `make_pooled`, which seems a little confusing. Mind fixing it?

Reply:
whoops! Fixed